examples/nodejs/token-vending-machine/infrastructure/lib/token-vending-machine-stack.ts

```diff
@@ -48,48 +49,145 @@ export class TokenVendingMachineStack extends cdk.Stack {
-        identitySources: [apig.IdentitySource.header('username'), apig.IdentitySource.header('password')],
-        resultsCacheTtl: cdk.Duration.seconds(5),
-      });
+        api.root.addCorsPreflight({
+          allowOrigins: ['*'],
+          allowHeaders: ['username', 'password']
+        });
 
-      api.root.addMethod('GET', tvmIntegration, {
-        authorizer: authorizer,
-      });
+        break;
+      }
+      case config.AuthenticationMethod.AmazonCognito: {
+        const userPool = new cognito.UserPool(this, 'MomentoTokenVendingMachineUserPool', {
+          userPoolName: 'MomentoTokenVendingMachineUserPool',
+          signInAliases: {
+            username: true
+          },
+          passwordPolicy: {
+            minLength: 8,
+            requireSymbols: true,
+          }
+        });
+        new cdk.CfnOutput(this, "UserPoolId", {
+          value: userPool.userPoolId,
+        });
+        
+        const userPoolClient = new cognito.UserPoolClient(this, 'MomentoTokenVendingMachineUserPoolClient', {
+          userPool,
+          generateSecret: false,
+          authFlows: {
+            userPassword: true
+          }
+        });
+        new cdk.CfnOutput(this, "UserPoolClientId", {
+          value: userPoolClient.userPoolClientId,
+        });
+
+        const authorizer = new apig.CognitoUserPoolsAuthorizer(this, 'MomentoTokenVendingMachineTokenAuthorizer', {
+          cognitoUserPools: [userPool],
+        });
+  
+        api.root.addMethod('GET', tvmIntegration, {
+          authorizationType: apig.AuthorizationType.COGNITO,
+          authorizer: authorizer,
+        });
+
+        api.root.addCorsPreflight({
+          allowOrigins: ['*'],
+        });
+
+        this.createCognitoUser(this, userPool, "momento", "$erverless");
+
+        break;
+      }
+      case config.AuthenticationMethod.Open: {
+        console.log("Warning: no authentication method provided, the Token Vending Machine URL will be publicly accessible");
+        api.root.addMethod('GET', tvmIntegration);
+        api.root.addCorsPreflight({
+          allowOrigins: ['*'],
+        });
+        break;
+      }
+      default: {
+        throw new Error("Unrecognized authentication method");
+      }
     }
-    else if (config.authenticationMethod === "amazon-cognito") {
-      const userPool = new cognito.UserPool(this, 'MomentoTokenVendingMachineUserPool', {
-        userPoolName: 'MomentoTokenVendingMachineUserPool',
-        signInAliases: {
-          username: true
+  }
+
+  // reference: https://github.com/awesome-cdk/cdk-userpool-user/blob/master/lib/UserPoolUser.ts 
+  private createCognitoUser(
+    scope: Construct,
+    userPool: cognito.IUserPool, 
+    username: string, 
+    password: string, 
+    group?: string
+  ) {
+    // Basically use the AWS SDK to fill in this gap in the CDK for creating a user with a password
+    const newUser = new AwsCustomResource(scope, 'AwsCustomResource-CreateCognitoUser', {
```

Comment:
wow, you made a custom resource! those have a pretty steep learning curve, nice job :)